### ISSUE TITLE
Excel uploading UX

### ISSUE THREAD

**Issue by petermarks-gsa** [GSA-TTS/FAC#449]
### Overview
Implement basic Excel template generation to be used screens like the federal awards screen. Should be componentized to be used on other screens. 

### Background

Users need to be able to upload and download Excel files associated with their submissions. (See [ADR](https://github.com/GSA-TTS/FAC/blob/main/docs/architecture/decisions/0015-store-uploaded-excel-files.md#validation).)

### Acceptance criteria: 
Happy path: 
* [ ] Go to screen with upload step (i.e. Federal Awards)
* [ ] Drag or choose valid local file and click "save & continue"
* [ ] The second and third steps are also checked off as completed.
* [ ] Should be able to replace file with replace button
* [ ] Excel files that pass the first [five validation levels](https://github.com/GSA-TTS/FAC/blob/main/docs/architecture/decisions/0015-store-uploaded-excel-files.md#validation) will have their contents extracted and stored in the database.
  * Blocked by data schema questions
- [ ] Other users on the same submission can download the file.

Error Path:
* [ ] Go to screen with upload step (i.e. Federal Awards)
* [ ] Drag or choose incorrectly formatted, empty, or incorrect local file and click "save & continue"
* [ ] You should see error messages as to why the file was rejected.
* [ ] Should be able to valid re-upload file and move forward

```[tasklist]
### Stories
- [x] #777 
- [ ] #778
- [ ] #779 
- [ ] #780 
- [ ] #781
- [ ] #782 
- [ ] #783 
- [ ] #784 
- [ ] https://github.com/GSA-TTS/FAC/issues/727
```

```[tasklist]
### Tasks
- [ ] #787 
- Excel download view
  - Associated access permissions.
  - Linking to the files from the view for the submission.
- frontend upload functionality / validation
- frontend download functionality
```

### Not in scope
- https://github.com/GSA-TTS/FAC/issues/443
- https://github.com/GSA-TTS/FAC/issues/450

**Comment by petermarks-gsa** [comment]
@melchoyce @austinhernandez To confirm my understanding: 

1. I assume the "replace" button in [this design](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=3159%3A20649) will show a warning (copy needed?), then show upload file dialog but not just "remove" the file? 

2. I assume that when the user returns to this screen after completing the step, they can replace the file. I assume we do not need to let them download it or show the name of the file they uploaded as I _believe_ the current FAC does not do this?

**Comment by melchoyce** [comment]
> * I assume the "replace" button in [this design](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=3159%3A20649) will show a warning (copy needed?), then show upload file dialog but not just "remove" the file?

Talked about this in design sync. We'll be using the USWDS file input component verbatim here:

https://designsystem.digital.gov/components/file-input/

**Comment by petermarks-gsa** [comment]
Also discussed at design sync: We do not have a design yet for downloading the uploaded file. While this is not part of the current Census product, we believe we need this as a way to see what you've already uploaded assuming we don't do a web form.

**Comment by melchoyce** [comment]
![image](https://user-images.githubusercontent.com/2846578/187553679-27335d00-d7c7-4416-9d16-b1d4f2c8d022.png)

@austinhernandez How do you feel about this placement? I figured we'd want it outside of the dropzone.

**Comment by melchoyce** [comment]
Update: We decided in cowork yesterday that we need to explore ways to make it more obvious a file has been uploaded, so another auditor can't come by and accidentally overwrite it. (Version control is currently out of scope.)

**Comment by melchoyce** [comment]
Worked with Austin and Erin earlier today on finalizing the upload/download interaction on this page:

![image](https://user-images.githubusercontent.com/2846578/189215083-e34f955c-edb1-4b10-bd55-5befbaf032d6.png)
[Figma mockups](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=4647%3A21220)

## Implementation Notes
- When a user downloads the worksheet:
  - The heading and button copy updates from “download” to “re-download”
  - The step counter is changed from a number to a checkmark.
- When you upload an incorrectly formatted, empty, or incorrect worksheet, the upload is rejected and you are given error messages as to why the file was rejected.
- When a user uploads a completed, correct worksheet:
  - The “re-download” button changes to some small explanation text with a download link.
  - The second step is also checked off as completed.
  - The filename is displyed alongside a “download file” action.

## Questions
- Is the excel file processed upon upload, or when a user clicks "save and continue?"

**Comment by melchoyce** [comment]
@petermarks-gsa The former:

> a generated excel template with no filled out info

Though if you upload a completely empty Excel document, that should also generate an error.

**Comment by jadudm** [comment]
Much/all of this has been picked up in other tickets. Closing this as a legacy ticket.
